huaweicloud/services/ram/resource_huaweicloud_ram_share.go

```diff
@@ -0,0 +1,479 @@
+		Schema: map[string]*schema.Schema{
+			"permission_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The permission id.`,
+			},
+			"permission_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The permission name.`,
+			},
+			"resource_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The resource type of the permission.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The status of the permission.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceRAMShareCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createRAMShare: create a RAM share.
+	var (
+		createRAMShareHttpUrl = "v1/resource-shares"
+		createRAMShareProduct = "ram"
+	)
+	createRAMShareClient, err := cfg.NewServiceClient(createRAMShareProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RAM Client: %s", err)
+	}
+
+	createRAMSharePath := createRAMShareClient.Endpoint + createRAMShareHttpUrl
+	createRAMShareOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	createRAMShareOpt.JSONBody = utils.RemoveNil(buildCreateRAMShareBodyParams(d))
+	createRAMShareResp, err := createRAMShareClient.Request("POST", createRAMSharePath, &createRAMShareOpt)
+	if err != nil {
+		return diag.Errorf("error creating RAM share: %s", err)
+	}
+
+	createRAMShareRespBody, err := utils.FlattenResponse(createRAMShareResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("resource_share.id", createRAMShareRespBody)
+	if err != nil {
+		return diag.Errorf("error creating RAMShare: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+	return resourceRAMShareRead(ctx, d, meta)
+}
+
+func buildCreateRAMShareBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":           utils.ValueIngoreEmpty(d.Get("name")),
+		"description":    utils.ValueIngoreEmpty(d.Get("description")),
+		"permission_ids": utils.ValueIngoreEmpty(d.Get("permission_ids").(*schema.Set).List()),
+		"principals":     utils.ValueIngoreEmpty(d.Get("principals").(*schema.Set).List()),
+		"resource_urns":  utils.ValueIngoreEmpty(d.Get("resource_urns").(*schema.Set).List()),
+		"tags":           utils.ExpandResourceTagsMap(d.Get("tags").(map[string]interface{})),
+	}
+	return bodyParams
+}
+
+func resourceRAMShareRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	shareClient, err := cfg.NewServiceClient("ram", region)
+	if err != nil {
+		return diag.Errorf("error creating RAM Client: %s", err)
+	}
+	// Search share instance information
+	if instanceDiag := setRAMShareInstance(shareClient, d); instanceDiag != nil {
+		return instanceDiag
+	}
```

Comment:
I think this will make the code easier to read
```suggestion
	if err:= setRAMShareInstance(shareClient, d); err != nil {
		return diag.FromErr(err)
	}
```